Kuzzle.Tests/KuzzleTest.cs

```diff
@@ -0,0 +1,182 @@
+using Xunit;
+using Newtonsoft.Json.Linq;
+using KuzzleSdk.Exceptions;
+using KuzzleSdk.Protocol;
+using KuzzleSdk.API;
+using System;
+using System.Threading.Tasks;
+using Moq;
+
+namespace Kuzzle.Tests {
+  public class KuzzleTest {
+    private readonly KuzzleSdk.Kuzzle _kuzzle;
+    private readonly Mock<AbstractProtocol> _protocol;
+
+    public KuzzleTest() {
+      _protocol = new Mock<AbstractProtocol>();
+      _kuzzle = new KuzzleSdk.Kuzzle(_protocol.Object);
+    }
+
+    [Fact]
+    public void DispatchTokenExpiredTest() {
+      bool eventDispatched = false;
+
+      _kuzzle.TokenExpired += delegate() {
+        eventDispatched = true;
+      };
+
+      _kuzzle.DispatchTokenExpired();
+
+      Assert.Null(_kuzzle.AuthenticationToken);
+      Assert.True(eventDispatched);
+    }
+
+    [Fact]
+    public void KuzzleConstructorTest() {
+      Assert.NotNull(_kuzzle.Auth);
+      Assert.NotNull(_kuzzle.Collection);
+      Assert.NotNull(_kuzzle.Document);
+      Assert.NotNull(_kuzzle.Realtime);
+      Assert.NotNull(_kuzzle.Server);
+
+      Assert.NotNull(_kuzzle.Version);
+      Assert.NotNull(_kuzzle.InstanceId);
+    }
+
+    [Fact]
+    public async void ConnectAsyncTest() {
+      await _kuzzle.ConnectAsync();
+
+      _protocol.Verify(m => m.ConnectAsync(), Times.Once);
+    }
+
+    [Fact]
+    public void DisconnectTest() {
+      _kuzzle.Disconnect();
+
+      _protocol.Verify(m => m.Disconnect(), Times.Once);
+    }
+
+
+    [Fact]
+    public void QueryAsyncTest() {
+      JObject request = new JObject {
+        { "controller", "test" },
+        { "action", "testAction" }
+      };
+      _kuzzle.AuthenticationToken = "jwt auth token";
+      _protocol.Setup(protocol => protocol.Send(It.IsAny<JObject>()));
+
+      _kuzzle.QueryAsync(request);
+
+      _protocol.Verify(
+        protocol => protocol.Send(It.Is<JObject>(o => testSendArg(o))));
+
+      Assert.Single(_kuzzle.requests);
+    }
+    private bool testSendArg (JObject query) {
+      Assert.Equal("test", query["controller"]);
+      Assert.Equal("testAction", query["action"]);
+      Assert.Equal("jwt auth token", query["jwt"]);
+      Assert.True(JToken.DeepEquals(query["volatile"], new JObject {
```

Comment:
(nitpicking) You can use `Assert.Equal` with JObject objects, by passing a JTokenEqualityComparer instance to it. This does the exact same thing, but is a bit easier to read.